### DISCUSSION TITLE
Distorted Images and tables

### DISCUSSION THREAD

**Discussion by lalitap25** [ImageMagick/ImageMagick#7355] [Help]
### ImageMagick version

7.1.1

### Operating system, version and so on

Linux

### Description

Hi There,

I have linux server for imagemagick and is used to convert the .emf or .wmf files to png format, but after the conversion the images get distorted for example from this 
<img width="251" alt="image" src="https://github.com/ImageMagick/ImageMagick/assets/148577211/28216aca-bf05-46f7-85de-4c3c2778573a">

to this
<img width="354" alt="image" src="https://github.com/ImageMagick/ImageMagick/assets/148577211/3fcd127b-f631-4a4d-bf64-ea9f0850fedb">

this is just an exampe of a table but we have issue with visio imgaes also
any idea??

**Comment by snibgo** [comment]
The image you show "from this" is a PNG file. It is not a EMF or WMF file.

Please show the actual input EMF or WMF file, and the command you used.

**Reply by lalitap25** [reply]
this is the error I get in the logs
Error output of executed command parameters [[/opt/ole-convert/oleconvert.sh, /tmp/picture3042988855418740275.wmf, /tmp/convertedPic10298416002904586266.png]]:

convert: FontSubstitutionRequired `Courier' @ warning/type.c/GetTypeInfoByFamily/490.

[Downloads.zip](https://github.com/user-attachments/files/15915335/Downloads.zip)
this zip has the wmf files.
this is the small script that is being used
###############################
#!/bin/sh
infile=$1
inname=`basename "$1"`
outfile=$2
outname=`basename "$2"`
LINUX_PATH=/polarion/magick-tmp
IMAGEMAGICK_LINUX_PATH=/visualization/magick-tmp
TARGET_SERVER='testuser@imagemagick_server'
cp "$infile" "$LINUX_PATH/$inname"
PWS_COMMAND="convert $IMAGEMAGICK_LINUX_PATH/$inname $IMAGEMAGICK_LINUX_PATH/$outname"
ssh $TARGET_SERVER $PWS_COMMAND
rm "$LINUX_PATH/$inname"
mv "$LINUX_PATH/$outname" "$outfile"

**Comment by amirjj** [comment]
Hi, I got problem for converting EMF to PNG in Linux how did you do that? is there any delegates to be added?
the problem explained in [this link](https://stackoverflow.com/questions/78792502/missingdelegateerror-no-decode-delegate-for-this-image-format-emf-converting-i)

**Comment by fmw42** [comment]
Look at the formats page at https://imagemagick.org/script/formats.php#supported.  It says on for Windows:


EMF	R	Microsoft Enhanced Metafile (32-bit)	Only available under Microsoft Windows. Use [-size](https://imagemagick.org/script/command-line-options.php#size) command line option to specify the maximum width and height.

**Reply by amirjj** [reply]
Thanks Fred, so my understanding is for Linux I need to use another library to convert `EMF` like using `LibreOffice` or `Inkscape`. Am I right? Do you suggest better solution?

**Reply by fmw42** [reply]
Sorry, I do not know. I am not a Windows or Linux user. I am on a Mac. I just looked on that page I referenced as saw the comment about Windows only.   But have you tried installing Inkspace to see if that works for you with Imagemagick?

**Comment by snibgo** [comment]
If LibreOffice or Inkscape can open your file, then you can create an entry in delegates.xml so IM will call that program to rasterize the image.

Inkscape can do this for a WMF that I have. I don't have any EMF files to test.
